### PR TITLE
Add visualization_msgs/Marker

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 Add visualization_msgs/Marker

## Summary

Add support for the visualization_msgs/Marker and visualization_msgs/MarkerArray types to the bridge.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: ros_ign_bridge/src/convert.cpp [hunk 1]
```diff
@@ -1237,4 +1261,252 @@ convert_ign_to_ros(
...
 
+template<>
+void
+convert_ros_to_ign(
+    const visualization_msgs::Marker & ros_msg,
+    ignition::msgs::Marker & ign_msg)
+{
+  convert_ros_to_ign(ros_msg.header, (*ign_msg.mutable_header()));
+
+  switch(ros_msg.action)
```
Comment (ahcorde) [inline ros_ign_bridge/src/convert.cpp:1274]: visualization_msgs::Marker::MODIFY ?

File: ros_ign_bridge/src/factories.cpp [hunk 2]
```diff
@@ -38,6 +38,17 @@ get_factory_impl(
...
+  if (
+    (ros_type_name == "std_msgs/ColorRGBA" || ros_type_name == "") &&
+     ign_type_name == "ignition.msgs.Color")
+  {
+    return std::make_shared<
+      Factory<
+        std_msgs::ColorRGBA,
+        ignition::msgs::Color
+      >
+    >("std_msgs/Bool", ign_type_name);
```
Comment (ahcorde) [inline ros_ign_bridge/src/factories.cpp]:
```suggestion
    >("std_msgs/ColorRGBA", ign_type_name);
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: ros_ign_bridge/src/convert.cpp (hunk 1 above)
Comment (mjcarroll) [reply ros_ign_bridge/src/convert.cpp:1274]: Good catch, thanks.

**Review by ahcorde — APPROVED** [review]
but CI is failing 

```
anyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

**Comment by mjcarroll** [comment]
> but CI is failing

I also remembered why I didn't have both `ADD` and `MODIFY` in there.  They map to the same value `0`.  You can't have multiple `cases` with the same value.  I'll revert that.

**Comment by ahcorde** [comment]
can you add a comment ?

**Comment by ahcorde** [comment]
@mjcarroll do we know how to proceed with Travis ?

**Comment by chapulina** [comment]
I re-triggered Travis, and now the build fails:


```
Errors << ros_ign_bridge:make /root/catkin_ws/logs/ros_ign_bridge/build.make.000.log

/root/catkin_ws/src/code/ros_ign_bridge/src/convert.cpp: In function 'void ros_ign_bridge::convert_ros_to_ign(const ROS_T&, IGN_T&) [with ROS_T = visualization_msgs::Marker_<std::allocator<void> >; IGN_T = ignition::msgs::Marker]':

/root/catkin_ws/src/code/ros_ign_bridge/src/convert.cpp:1299:48: error: 'ARROW' is not a member of 'ignition::msgs::Marker'

       ign_msg.set_type(ignition::msgs::Marker::ARROW);

                                                ^~~~~

/root/catkin_ws/src/code/ros_ign_bridge/src/convert.cpp: In function 'void ros_ign_bridge::convert_ign_to_ros(const IGN_T&, ROS_T&) [with ROS_T = visualization_msgs::Marker_<std::allocator<void> >; IGN_T = ignition::msgs::Marker]':

/root/catkin_ws/src/code/ros_ign_bridge/src/convert.cpp:1438:34: error: 'CONE' is not a member of 'ignition::msgs::Marker'

     case ignition::msgs::Marker::CONE:

                                  ^~~~

/root/catkin_ws/src/code/ros_ign_bridge/src/convert.cpp:1442:34: error: 'ARROW' is not a member of 'ignition::msgs::Marker'

     case ignition::msgs::Marker::ARROW:

                                  ^~~~~

/root/catkin_ws/src/code/ros_ign_bridge/src/convert.cpp:1445:34: error: 'AXIS' is not a member of 'ignition::msgs::Marker'

     case ignition::msgs::Marker::AXIS:

                                  ^~~~
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
CI is failing

**Comment by mjcarroll** [comment]
Ah, so it looks like the difference is `citadel` vs `dome`, as the message has changed between the two releases.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**